include/kernel/thread.h

```diff
@@ -171,7 +173,25 @@ struct _thread_userspace_local_data {
 
 typedef struct k_thread_runtime_stats {
 #ifdef CONFIG_SCHED_THREAD_USAGE
+	/*
+	 * In the context of thread statistics, [execution_cycles] is the same
+	 * as the total # of non-idle cycles. In the context of CPU statistics,
+	 * it refers to the sum of non-idle + idle cycles.
+	 */
+
 	uint64_t execution_cycles;
+	uint64_t current_cycles;      /* current # of non-idle cycles */
+	uint64_t peak_cycles;         /* peak # of non-idle cycles */
+	uint64_t total_cycles;        /* total # of non-idle cycles */
+	uint64_t average_cycles;      /* average # of non-idle cycles */
```

Comment:
Seems like lots of apps will want to trace cycle usage.  Probably far fewer will want 40 bytes of extra analysis in their thread structs?  Seems like this wants to be under a CONFIG_SCHED_THREAD_USAGE_ANALYSIS or something.

Reply:
I like this idea.